src/slack_interaction_handler.ts

```diff
@@ -319,3 +344,201 @@ export async function handleVolunteerUpdate({
     newBlocks: payload.message.blocks,
   });
 }
+
+// This function receives the initial request to reset a demo
+// and response by creating a modal populated with data needed
+// to reset the demo if the Slack user confirms.
```

Comment:
Maybe I'm missing it, but is there (should there be?) an admin check in this function? Otherwise it seems like anyone can reset demos?

Reply:
Thanks for helping me think about this! For this particular command, we want all volunteers to be able to use it to reset their own demo conversations. With regard to any _particular_ demo conversation, right now only those Slack users who have access to the conversation can reset it, which should only be admins, the volunteer in the demo conversation and their podmates. I think that's perfect in this case. And if someone selects Reset Demo from a thread that is no longer active or tries to Reset Demo on a non-demo conversation, they will get an error modal. What do you think / does that sound right to you? Really appreciate you helping me think about this as I've had my head down in the weeds on this for a while.

With regard to permissions for the other commands we have lined up:
-**Route Voter To Channel:** should be a Shortcut that only works for admins
-**Opening Channels (however we implement it):** only available to admins
-**Route Voter to Journey:** should have the same access control as Reset Voter